### PR TITLE
ref: cleanup configuration and localization controller

### PR DESCRIPTION
**Pull request by Skarlso**
This change covers the following changes:
- Unify the main control loop for the two controllers
- Unify the Spec and Status of the two controllers

Coming up:
- Refactor the test to cover the mutation controller and reduce the tests for the other controllers.

### CODE REVIEW

**Review by jmickey — COMMENTED** [review]
LGTM

File: api/v1alpha1/mutation_types.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import (
+	"time"
+
+	"github.com/fluxcd/pkg/apis/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// MutationSpec defines a common spec between Localization and Configuration.
+type MutationSpec struct {
```
Comment (phoban01) [inline api/v1alpha1/mutation_types.go:12]: mutation is a good term 👍🏻

Comment (phoban01) [reply api/v1alpha1/mutation_types.go:12]: Should we add componentversion as part of the `MutationSpec`? Currently we are taking it from the `ConfigRef`  which is not right, that should only be for specify the component from which to fetch the config.

I think` spec.source` should support either kind=Source or kind=ComponentVersion; that is to say if the source is not a snapshot CR then it should be a componentversion CR.

Comment (Skarlso) [reply api/v1alpha1/mutation_types.go:12]: I agree. But let's do that in a separate PR. I feel like that could get bigger. :D

**Review by jmickey — APPROVED** [review]

**Review by phoban01 — APPROVED** [review]

**Pull request merged**